src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx

```diff
@@ -218,6 +227,13 @@ export class ProsemirrorAdapter {
     private handleTransaction = (transaction) => {
         const newState = this.view.state.apply(transaction);
         this.view.updateState(newState);
+
+        if (isEqual(this.prosemirrorDoc, transaction.doc) && this.shouldEmit) {
+            this.shouldEmit = false;
```

Comment:
I think the naming is a little bit confusing here. We seem to only be avoiding emitting something if `this.shouldEmit` is _true_, and then we set it to false, so it will never be true again…?

Reply:
Fixed

Reply:
I'm not completely at peace with the fixup you made for this, because we now instead have a flag that says `this.shouldEmit = false` and we're still emitting stuff unless a specific circumstance happens. So I think I'll remove that fixup while I'm fiddling around with some stuff, and then try to find a different name for the property or something.

I'm going to fiddle around with the commits a little bit, so there may be merge conflicts if you are making changes at the same time.

Reply:
I'm not sure I completely understand how this if statement works 😅

This part specifically:

```ts
isEqual(this.prosemirrorDoc, transaction.doc)
```

As far as I understand, we want to avoid emitting a `change` event if the changes are only syntactic changes that prosemirror makes automatically, stuff like replacing `<b>` with `<strong>`, for example. But I don't understand what `this.prosemirrorDoc` and `transaction.doc` are, or how their values being equal to each other tells us what we need to know? 🤔